### DISCUSSION TITLE
how to use WeightRandomSampler in mmclassification

### DISCUSSION THREAD

**Discussion by ChenJunzhi-buaa** [open-mmlab/mmclassification#1326] [Q&A]
How to rewrite a WeightRandomSampler Class in mmclassification‘s SAMPLERS ?

**Comment by ChenJunzhi-buaa** [comment]
> How to rewrite a WeightRandomSampler Class in mmclassification‘s SAMPLERS ?

What if only inherit from torch.utils.data.WeightRandomSampler without any other changes?

**Reply by Ezra-Yu** [reply]
> What if only inherit from torch.utils.data.WeightRandomSampler without any other changes?

It can be done that way.

**Reply by ChenJunzhi-buaa** [reply]
Thanks! I Write it like below. There is no run error. But the 5 five necessary inputs are not used. Will some other errors be caused by such incomplete codes？
![image](https://user-images.githubusercontent.com/77711950/215309980-0cf22c42-bdd6-46e8-91cc-4644c6b31989.png)

**Reply by Ezra-Yu** [reply]
Sorry，I haven't read the source code for it, so I can't guarantee that. What the relevant parameters do would require you to check the documentation or code to confirm.
